ext/opentelemetry-ext-otcollector/src/opentelemetry/ext/otcollector/trace_exporter/__init__.py

```diff
@@ -0,0 +1,158 @@
+
+"""OpenTelemetry Collector Exporter."""
+
+import logging
+from typing import Optional, Sequence
+
+import grpc
+from opencensus.proto.agent.trace.v1 import (
+    trace_service_pb2,
+    trace_service_pb2_grpc,
+)
+from opencensus.proto.trace.v1 import trace_pb2
+
+import opentelemetry.ext.otcollector.util as utils
+from opentelemetry.sdk.trace import Span, SpanContext
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.trace import SpanKind, TraceState
+
+DEFAULT_ENDPOINT = "http://localhost:55678"
+
+logger = logging.getLogger(__name__)
+
+
+# pylint: disable=no-member
+class CollectorSpanExporter(SpanExporter):
+    """OpenTelemetry Collector span exporter.
+
+    Args:
+        endpoint: OpenTelemetry Collector OpenCensus receiver endpoint.
+        service_name: Name of Collector service.
+        host_name: Host name.
+        client: TraceService client stub.
+    """
+
+    def __init__(
+        self, endpoint=None, service_name=None, host_name=None, client=None
+    ):
+        self.endpoint = DEFAULT_ENDPOINT if endpoint is None else endpoint
+
+        if client is None:
+            self.channel = grpc.insecure_channel(self.endpoint)
+            self.client = trace_service_pb2_grpc.TraceServiceStub(
+                channel=self.channel
+            )
+        else:
+            self.client = client
+
+        self.node = utils.get_node(service_name, host_name)
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        collector_spans = translate_to_collector(spans)
+        export_request = trace_service_pb2.ExportTraceServiceRequest(
+            node=self.node, spans=collector_spans
+        )
+        try:
+            self.client.Export(export_request)
+        except grpc.RpcError:
+            return SpanExportResult.FAILED_NOT_RETRYABLE
+
+        return SpanExportResult.SUCCESS
+
+    def shutdown(self) -> None:
+        pass
+
+
+# pylint: disable=too-many-branches
+def translate_to_collector(spans: Sequence[Span]):
+    collector_spans = []
+    for span in spans:
+        collector_span = trace_pb2.Span(
+            name=trace_pb2.TruncatableString(value=span.name),
+            kind=utils.get_collector_span_kind(span.kind),
+            trace_id=span.context.trace_id.to_bytes(16, "big"),
+            span_id=span.context.span_id.to_bytes(8, "big"),
+            start_time=utils.proto_timestamp_from_time_ns(span.start_time),
+            end_time=utils.proto_timestamp_from_time_ns(span.end_time),
+            status=trace_pb2.Status(
+                code=span.status.canonical_code.value,
+                message=span.status.description,
+            )
+            if span.status is not None
+            else None,
+        )
+
+        if span.parent is not None:
+            collector_span.parent_span_id = span.parent.span_id.to_bytes(
+                8, "big"
+            )
+
+        if span.context.trace_state is not None:
```

Comment:
I believed I triggered this one with a manually created Span in a unit test, same as others checks in this method I added if x instead of if x is not None.